processor/filterprocessor/filter_processor.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filterprocessor
+
+import (
+	"context"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumerdata"
+	"go.opentelemetry.io/collector/internal/processor/filtermetric"
+)
+
+type filterMetricProcessor struct {
+	cfg          *Config
+	capabilities component.ProcessorCapabilities
+	next         consumer.MetricsConsumerOld
+	include      *filtermetric.Matcher
+	exclude      *filtermetric.Matcher
+}
+
+var _ component.MetricsProcessorOld = (*filterMetricProcessor)(nil)
+
+func newFilterMetricProcessor(next consumer.MetricsConsumerOld, cfg *Config) (*filterMetricProcessor, error) {
+	inc, err := createMatcher(cfg.Metrics.Include)
+	if err != nil {
+		return nil, err
+	}
+
+	exc, err := createMatcher(cfg.Metrics.Exclude)
+	if err != nil {
+		return nil, err
+	}
+
+	return &filterMetricProcessor{
+		cfg:          cfg,
+		capabilities: component.ProcessorCapabilities{MutatesConsumedData: false},
+		next:         next,
+		include:      inc,
+		exclude:      exc,
+	}, nil
+}
+
+func createMatcher(mp *filtermetric.MatchProperties) (*filtermetric.Matcher, error) {
+	// Nothing specified in configuration
+	if mp == nil {
+		return nil, nil
+	}
+
+	matcher, err := filtermetric.NewMatcher(mp)
+	if err != nil {
+		return nil, err
+	}
+
+	return &matcher, nil
+}
+
+// GetCapabilities returns the Capabilities assocciated with the resource processor.
+func (fmp *filterMetricProcessor) GetCapabilities() component.ProcessorCapabilities {
+	return fmp.capabilities
```

Comment:
no need to cache capabilities, it is called only during start time.

Reply:
good to know!